### PR TITLE
chore: update gateway-addons-helm

### PR DESCRIPTION
**Pull request by zirain**
- disable grafana-agent and loki canary
- merge and clean makefile

### CODE REVIEW

**Comment by zirain** [comment]
wait https://github.com/envoyproxy/gateway/pull/3610

**Review by shawnh2 — APPROVED** [review]
LGTM thanks

File: tools/make/kube.mk [hunk 1]
```diff
@@ -141,7 +141,7 @@ endif
 .PHONY: install-e2e-telemetry
 install-e2e-telemetry: helm-generate.gateway-addons-helm
 	@$(LOG_TARGET)
-	helm install eg-addons charts/gateway-addons-helm --set grafana.enabled=false,opentelemetry-collector.enabled=true -n monitoring --create-namespace --timeout='$(WAIT_TIMEOUT)' --wait --wait-for-jobs
+	helm upgrade -i eg-addons charts/gateway-addons-helm --set grafana.enabled=false,opentelemetry-collector.enabled=true -n monitoring --create-namespace --timeout='$(WAIT_TIMEOUT)' --wait --wait-for-jobs
```
Comment (arkodg) [inline tools/make/kube.mk:144]: curious why this was done

**Pull request merged**